L1Trigger/L1THGCal/test/HGCalTriggerGeomTesterV9Imp3.cc

```diff
@@ -0,0 +1,1397 @@
+          edm::LogProblem("BadStage1") << "Error: \n Module " << module << "(" << HGCalModuleDetId(module)
+                                       << ")\n has not been found in \n stage-1 " << HGCalBackendDetId(stage1);
+          std::stringstream output;
+          output << "   Available modules are:\n";
+          for (auto module_geom : modules_geom) {
+            output << module_geom << " ";
+          }
+          output << "   Connected lpgbts are:\n";
+          for (auto lpgbt : lpgbts) {
+            output << lpgbt << " ";
+          }
+          edm::LogProblem("BadStage1") << output.str();
+          throw cms::Exception("BadGeometry")
+              << "HGCalTriggerGeometry: Found inconsistency in Stage1 <-> module mapping";
+        }
+      }
+    }
+
+    // Filling Stage 2 FPGA -> Stage 1 FPGA
+
+    edm::LogPrint("ModuleCheck") << "Checking Stage 1 -> Stage 2 -> Stage 1 consistency";
+    trigger_map_set stage2_to_stage1;
+    for (const auto& stage1 : stage1_to_modules) {
+      HGCalBackendDetId id(stage1.first);
+      HGCalTriggerGeometryBase::geom_set stage2FPGAs = triggerGeometry_->getStage2FpgasFromStage1Fpga(id);
+      for (const auto& stage2 : stage2FPGAs) {
+        auto itr_insert = stage2_to_stage1.emplace(stage2, std::unordered_set<uint32_t>());
+        itr_insert.first->second.emplace(id);
+      }
+    }
+    // checking S1 -> S2 consistency
+
+    for (const auto& stage2_modules : stage2_to_stage1) {
+      HGCalBackendDetId stage2(stage2_modules.first);
+
+      // Check consistency of Stage-1 FPGA going to Stage 2 FPGA
+      HGCalTriggerGeometryBase::geom_set stage1FPGAs = triggerGeometry_->getStage1FpgasFromStage2Fpga(stage2);
+
+      const auto& stage1fpgas = stage2_modules.second;
+
+      for (auto stage1fpga : stage1fpgas) {
+        if (stage1FPGAs.find(stage1fpga) == stage1FPGAs.end()) {
+          edm::LogProblem("BadStage2") << "Error: \n Stage-1 FPGA " << stage1fpga << "("
+                                       << HGCalBackendDetId(stage1fpga) << ")\n has not been found in \n stage-2 "
+                                       << HGCalBackendDetId(stage2);
+          std::stringstream output;
+          output << "   Available Stage-1 FPGAs are:\n";
+          for (auto stage1FPGA : stage1FPGAs) {
+            output << stage1FPGA << " ";
+          }
+          edm::LogProblem("BadStage2") << output.str();
+          throw cms::Exception("BadGeometry")
+              << "HGCalTriggerGeometry: Found inconsistency in Stage2 <-> Stage1 mapping";
+        }
+      }
+    }
+
+  } catch (const cms::Exception& e) {
+    edm::LogWarning("HGCalTriggerGeometryTester")
+        << "Problem with the trigger geometry detected. Only the basic cells tree will be filled\n";
+    edm::LogWarning("HGCalTriggerGeometryTester") << e.message() << "\n";
+    return false;
+  }
+  return true;
+}
+
+bool HGCalTriggerGeomTesterV9Imp3::checkNeighborConsistency() {
```

Comment:
This function can actually be removed as there is no neighbor definition

Reply:
thanks, removed